TECHNICAL-SETUP.md

```diff
@@ -0,0 +1,67 @@
+# Technical setup 
+
+
+## Software requirement
+
+To build the project:
+
+- [Golang](https://golang.org/dl/) v1.16 or greater 
+- [`make`](https://www.gnu.org/software/make/) - to use Makefile targets
+- `sha256sum` = to calculate build checksums
+
+To run the [seed scripts](scripts/seeds/README.md)
+
+- OS: linux or macOS
+- [jq](https://stedolan.github.io/jq/) - json processor
+
+For git pre-commit hooks
+
+- python
+- [pre-commit](https://pre-commit.com/#install) - git pre-commit hooks framework 
+- [golangci-lint](https://github.com/golangci/golangci-lint) - linter for Go 
+
+### Swagger 
+
+To generate the swagger UI the following dependencies must be installed
+
+- [`swagger-combine`](https://www.npmjs.com/package/swagger-combine)
+
+
+## Development environment setup (only for contributing)
+
+For development it is highly recommended to install the configured pre-commit hooks by running from the project's root folder:
+
+```sh
+$ pre-commit install
+$ pre-commit install --hook-type commit-msg
+```
+
+This operations should be run only once. 
+
+## Building 
+
+To build the project, run the `make build` command from the project's root folder. The output of the build will be generated in the `build` folder.
+
+For cross-builds use the standard `GOOS` and `GOARCH` env vars. i.e. to build for windows:
```

Comment:
```suggestion
## Building the project

To build the project, run the `make build` command from the project's root folder. The output of the build is generated in the `build` folder.

For cross-builds, use the standard `GOOS` and `GOARCH` environment variables. For example, to build the project for Windows:
```

Reply:
so tell us what "the project" is here. And why we want to build for another OS?

Reply:
we may want to build for another os to distribute the app to another party, both in a CI system of to distribute it to a 3rd party that doesn't have a build environment